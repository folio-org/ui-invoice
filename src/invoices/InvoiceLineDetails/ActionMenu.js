import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import {
  Button,
  Icon,
  MenuSection,
} from '@folio/stripes/components';
import { IfPermission } from '@folio/stripes/core';

const ActionMenu = ({ onDelete, onEdit, isDeletable }) => (
  <MenuSection id="invoice-line-details-actions">
    <IfPermission perm="invoice.invoice-lines.item.put">
      <Button
        data-testid="invoice-line-edit"
        buttonStyle="dropdownItem"
        data-test-button-edit-invoice-line
        onClick={onEdit}
      >
        <Icon size="small" icon="edit">
          <FormattedMessage id="ui-invoice.button.edit" />
        </Icon>
      </Button>
    </IfPermission>
    {isDeletable && (
      <IfPermission perm="invoice.invoice-lines.item.delete">
        <Button
          data-testid="invoice-line-delete"
          buttonStyle="dropdownItem"
          data-test-button-delete-invoice-line
          onClick={onDelete}
        >
          <Icon size="small" icon="trash">
            <FormattedMessage id="ui-invoice.button.delete" />
          </Icon>
        </Button>
      </IfPermission>
    )}
  </MenuSection>
);

ActionMenu.propTypes = {
  onDelete: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  isDeletable: PropTypes.bool.isRequired,
};

export default ActionMenu;
