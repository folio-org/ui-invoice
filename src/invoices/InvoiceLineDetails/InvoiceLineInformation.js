import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Checkbox,
  Col,
  KeyValue,
  NoValue,
  Row,
  TextLink,
} from '@folio/stripes/components';
import {
  ClipCopy,
  ViewMetaData,
} from '@folio/stripes/smart-components';
import {
  AmountWithCurrencyField,
  FolioFormattedDate,
  VendorReferenceNumbersDetails,
} from '@folio/stripes-acq-components';

import { CalculatedExchangeAmount } from '../../common/components';

const InvoiceLineInformation = ({
  currency,
  exchangeRate,
  invoiceLine,
  poLine,
}) => {
  const metadata = invoiceLine?.metadata;

  return (
    <>
      {metadata && <ViewMetaData metadata={metadata} />}
      <Row>
        <Col xs={12}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.description" />}
            value={invoiceLine?.description}
          />
        </Col>
        <Col xs={3}>
          <KeyValue label={<FormattedMessage id="ui-invoice.invoiceLine.poLineNumber" />}>
            {poLine?.id
              ? (
                <>
                  <TextLink
                    rel="noopener noreferrer"
                    target="_blank"
                    to={`/orders/lines/view/${poLine.id}`}
                  >
                    {poLine.poLineNumber}
                  </TextLink>
                  <ClipCopy text={poLine.poLineNumber} />
                </>
              )
              : <NoValue />
            }
          </KeyValue>
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.invoiceLineNumber" />}
            value={invoiceLine?.invoiceLineNumber}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoice.details.information.status" />}
            value={invoiceLine?.invoiceLineStatus}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.subscriptionInfo" />}
            value={invoiceLine?.subscriptionInfo}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.subscriptionStart" />}
            value={<FolioFormattedDate value={invoiceLine?.subscriptionStart} />}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.subscriptionEnd" />}
            value={<FolioFormattedDate value={invoiceLine?.subscriptionEnd} />}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.comment" />}
            value={invoiceLine?.comment}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.accountNumber" />}
            value={invoiceLine?.accountNumber}
          />
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoice.accountingCode" />}
            value={invoiceLine?.accountingCode}
          />
        </Col>
        <Col xs={3}>
          <KeyValue label={<FormattedMessage id="ui-invoice.invoiceLine.subTotal" />}>
            <AmountWithCurrencyField
              amount={invoiceLine?.subTotal}
              currency={currency}
            />
          </KeyValue>
        </Col>
        <Col xs={3}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.quantity" />}
            value={invoiceLine?.quantity}
          />
        </Col>
        <Col xs={3}>
          <CalculatedExchangeAmount
            currency={currency}
            exchangeRate={exchangeRate}
            label={<FormattedMessage id="ui-invoice.invoice.details.lines.list.total.exchanged" />}
            total={invoiceLine?.total}
          />
        </Col>
        <Col xs={3}>
          <Checkbox
            checked={Boolean(invoiceLine?.releaseEncumbrance)}
            disabled
            label={<FormattedMessage id="ui-invoice.invoiceLine.releaseEncumbrance" />}
            type="checkbox"
            vertical
          />
        </Col>
        <Col xs={12}>
          <KeyValue
            label={<FormattedMessage id="ui-invoice.invoiceLine.referenceNumbers" />}
          >
            <VendorReferenceNumbersDetails referenceNumbers={invoiceLine?.referenceNumbers} />
          </KeyValue>
        </Col>
      </Row>
    </>
  );
};

InvoiceLineInformation.propTypes = {
  currency: PropTypes.string,
  exchangeRate: PropTypes.number,
  invoiceLine: PropTypes.object,
  poLine: PropTypes.object,
};

export default InvoiceLineInformation;
