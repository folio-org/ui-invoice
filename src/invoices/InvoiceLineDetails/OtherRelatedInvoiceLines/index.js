export { OtherRelatedInvoiceLines } from './OtherRelatedInvoiceLines';
export { useOtherRelatedInvoiceLines } from './useOtherRelatedInvoiceLines';
