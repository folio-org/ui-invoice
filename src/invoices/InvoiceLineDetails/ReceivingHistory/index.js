export { default as ReceivingHistory } from './ReceivingHistory';
