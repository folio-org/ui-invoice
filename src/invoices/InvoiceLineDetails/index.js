export { default as InvoiceLineDetails } from './InvoiceLineDetails';
export { default as InvoiceLineDetailsContainer } from './InvoiceLineDetailsContainer';
