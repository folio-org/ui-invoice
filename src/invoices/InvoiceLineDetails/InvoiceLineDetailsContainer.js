import React, { useCallback, useEffect, useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';

import { LoadingPane } from '@folio/stripes/components';
import {
  baseManifest,
  Tags,
  useModalToggle,
  useShowCallout,
} from '@folio/stripes-acq-components';
import { stripesConnect } from '@folio/stripes/core';

import {
  invoiceLineResource,
  invoiceResource,
} from '../../common/resources';
import { PO_LINES_API } from '../../common/constants';
import InvoiceLineDetails from './InvoiceLineDetails';

const InvoiceLineDetailsContainer = ({
  history,
  location,
  match: { params },
  mutator: originMutator,
}) => {
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const mutator = useMemo(() => originMutator, []);
  const [polNumber, setPolNumber] = useState('');
  const [invoice, setInvoice] = useState({});
  const [invoiceLine, setInvoiceLine] = useState({});
  const [isLoading, setIsLoading] = useState(true);
  const [isTagsPaneOpened, setTagsPaneOpened] = useModalToggle();
  const showCallout = useShowCallout();

  const fetchInvoiceLineDetails = useCallback(
    () => {
      const invoicePromies = mutator.invoice.GET().then(response => setInvoice(response));
      const invoiceLinePromise = mutator.invoiceLine.GET()
        .then(line => {
          setInvoiceLine(line);

          return line.poLineId && mutator.poLine.GET({ path: `${PO_LINES_API}/${line.poLineId}` });
        })
        .then(poLine => setPolNumber(poLine?.poLineNumber || ''));

      return Promise.all([invoicePromies, invoiceLinePromise]);
    },
    [mutator.invoice, mutator.invoiceLine, mutator.poLine],
  );

  useEffect(() => {
    setIsLoading(true);
    fetchInvoiceLineDetails().finally(setIsLoading);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [params.id]);

  const updateInvoiceLineTagList = async (line) => {
    setIsLoading(true);
    await mutator.invoiceLine.PUT(line);
    await fetchInvoiceLineDetails();
    setIsLoading(false);
  };

  const closeInvoiceLine = useCallback(
    () => {
      const pathname = `/invoice/view/${params.id}`;

      history.push({
        pathname,
        search: location.search,
      });
    },
    [params.id, history, location.search],
  );

  const goToEditInvoiceLine = useCallback(
    () => {
      const pathname = `/invoice/view/${params.id}/line/${params.lineId}/edit`;

      history.push({
        pathname,
        search: location.search,
      });
    },
    [params.id, params.lineId, history, location.search],
  );

  const deleteInvoiceLine = useCallback(
    () => {
      setIsLoading(true);
      mutator.invoiceLine.DELETE({ id: params.lineId }, { silent: true })
        .then(() => {
          showCallout({ messageId: 'ui-invoice.invoiceLine.hasBeenDeleted' });
          history.replace({
            pathname: `/invoice/view/${params.id}`,
            search: location.search,
          });
        })
        .catch(() => {
          setIsLoading(false);
          showCallout({ messageId: 'ui-invoice.errors.invoiceLineHasNotBeenDeleted', type: 'error' });
        });
    },
    [mutator.invoiceLine, params.lineId, params.id, showCallout, history, location.search],
  );

  if (isLoading || invoiceLine?.id !== params.lineId) {
    return (
      <LoadingPane dismissible onClose={closeInvoiceLine} />
    );
  }

  return (
    <>
      <InvoiceLineDetails
        closeInvoiceLine={closeInvoiceLine}
        currency={invoice.currency}
        deleteInvoiceLine={deleteInvoiceLine}
        goToEditInvoiceLine={goToEditInvoiceLine}
        invoiceLine={invoiceLine}
        poLineNumber={polNumber}
        tagsToggle={setTagsPaneOpened}
      />
      {isTagsPaneOpened && (
        <Tags
          putMutator={updateInvoiceLineTagList}
          recordObj={invoiceLine}
          onClose={setTagsPaneOpened}
        />
      )}
    </>
  );
};

InvoiceLineDetailsContainer.manifest = Object.freeze({
  invoiceLine: {
    ...invoiceLineResource,
    accumulate: true,
    fetch: false,
  },
  invoice: {
    ...invoiceResource,
    accumulate: true,
    fetch: false,
  },
  poLine: {
    ...baseManifest,
    path: PO_LINES_API,
    accumulate: true,
    fetch: false,
  },
});

InvoiceLineDetailsContainer.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  match: ReactRouterPropTypes.match.isRequired,
  mutator: PropTypes.object.isRequired,
};

export default stripesConnect(InvoiceLineDetailsContainer);
