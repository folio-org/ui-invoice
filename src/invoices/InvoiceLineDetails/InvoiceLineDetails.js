import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { get } from 'lodash';

import {
  Accordion,
  AccordionSet,
  Col,
  ConfirmationModal,
  ExpandAllButton,
  Pane,
  Row,
} from '@folio/stripes/components';

import {
  FundDistributionView,
} from '@folio/stripes-acq-components';

import TagsBadge from '../Tags/TagsBadge';
import ActionMenu from './ActionMenu';
import InvoiceLineInformation from './InvoiceLineInformation';
import {
  SECTIONS_INVOICE_LINE,
} from '../constants';
import {
  expandAll,
  toggleSection,
} from '../../common/utils';
import AdjustmentsDetails from '../AdjustmentsDetails';

class InvoiceLineDetails extends Component {
  static propTypes = {
    closeInvoiceLine: PropTypes.func.isRequired,
    deleteInvoiceLine: PropTypes.func.isRequired,
    goToEditInvoiceLine: PropTypes.func.isRequired,
    invoiceLine: PropTypes.object.isRequired,
    tagsToggle: PropTypes.func.isRequired,
  };

  constructor(props, context) {
    super(props, context);
    this.expandAll = expandAll.bind(this);
    this.toggleSection = toggleSection.bind(this);
  }

  state = {
    sections: {},
    showConfirmDelete: false,
  };

  renderActionMenu = ({ onToggle }) => {
    const { goToEditInvoiceLine } = this.props;

    return (
      <ActionMenu
        onEdit={() => {
          onToggle();
          goToEditInvoiceLine();
        }}
        onDelete={() => {
          onToggle();
          this.mountDeleteLineConfirm();
        }}
      />
    );
  }

  mountDeleteLineConfirm = () => this.setState({ showConfirmDelete: true });

  unmountDeleteConfirm = () => this.setState({ showConfirmDelete: false });

  render() {
    const {
      closeInvoiceLine,
      deleteInvoiceLine,
      invoiceLine,
      tagsToggle,
    } = this.props;
    const { sections, showConfirmDelete } = this.state;
    const { invoiceLineNumber, adjustments } = invoiceLine;
    const tags = get(invoiceLine, ['tags', 'tagList'], []);
    const fundDistributions = get(invoiceLine, 'fundDistributions', []);
    const total = get(invoiceLine, 'total', 0);

    const paneTitle = (
      <FormattedMessage
        id="ui-invoice.invoiceLine.paneTitle.view"
        values={{ invoiceLineNumber }}
      />
    );

    const lastMenu = (
      <TagsBadge
        tagsToggle={tagsToggle}
        tagsQuantity={tags.length}
      />
    );

    return (
      <Pane
        id="pane-invoiceLineDetails"
        defaultWidth="fill"
        dismissible
        onClose={closeInvoiceLine}
        paneTitle={paneTitle}
        actionMenu={this.renderActionMenu}
        lastMenu={lastMenu}
      >
        <Row end="xs">
          <Col xs={12}>
            <ExpandAllButton
              accordionStatus={sections}
              onToggle={this.expandAll}
            />
          </Col>
        </Row>
        <AccordionSet
          accordionStatus={sections}
          onToggle={this.toggleSection}
        >
          <Accordion
            id={SECTIONS_INVOICE_LINE.invoiceLineInformation}
            label={<FormattedMessage id="ui-invoice.invoiceLineInformation" />}
          >
            <InvoiceLineInformation invoiceLine={invoiceLine} />
          </Accordion>
          <Accordion
            id={SECTIONS_INVOICE_LINE.fundDistribution}
            label={<FormattedMessage id="ui-invoice.fundDistribution" />}
          >
            <FundDistributionView
              fundDistributions={fundDistributions}
              totalAmount={total}
            />
          </Accordion>
          <Accordion
            id={SECTIONS_INVOICE_LINE.adjustments}
            label={<FormattedMessage id="ui-invoice.adjustments" />}
          >
            <AdjustmentsDetails adjustments={adjustments} />
          </Accordion>
        </AccordionSet>
        {showConfirmDelete && (
          <ConfirmationModal
            id="delete-invoice-line-confirmation"
            confirmLabel={<FormattedMessage id="ui-invoice.invoiceLine.delete.confirmLabel" />}
            heading={<FormattedMessage id="ui-invoice.invoiceLine.delete.heading" values={{ invoiceLineNumber }} />}
            message={<FormattedMessage id="ui-invoice.invoiceLine.delete.message" />}
            onCancel={this.unmountDeleteConfirm}
            onConfirm={deleteInvoiceLine}
            open
          />
        )}
      </Pane>
    );
  }
}

export default InvoiceLineDetails;
