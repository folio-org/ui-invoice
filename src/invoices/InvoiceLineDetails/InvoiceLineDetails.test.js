import { MemoryRouter } from 'react-router-dom';

import {
  act,
  render,
  screen,
} from '@folio/jest-config-stripes/testing-library/react';
import user from '@folio/jest-config-stripes/testing-library/user-event';
import {
  HasCommand,
  expandAllSections,
  collapseAllSections,
} from '@folio/stripes/components';

import { useExchangeCalculation } from '@folio/stripes-acq-components';

import {
  invoiceLine,
  orderLine,
} from 'fixtures';
import ActionMenu from './ActionMenu';
import InvoiceLineDetails from './InvoiceLineDetails';

jest.mock('@folio/stripes-components/lib/Commander', () => ({
  HasCommand: jest.fn(({ children }) => <div>{children}</div>),
  expandAllSections: jest.fn(),
  collapseAllSections: jest.fn(),
}));
jest.mock('@folio/stripes-acq-components', () => ({
  ...jest.requireActual('@folio/stripes-acq-components'),
  FundDistributionView: jest.fn(() => 'FundDistributionView'),
  useExchangeCalculation: jest.fn(),
}));

jest.mock('../AdjustmentsDetails', () => jest.fn().mockReturnValue('AdjustmentsDetails'));
jest.mock('./OtherRelatedInvoiceLines', () => ({
  OtherRelatedInvoiceLines: jest.fn().mockReturnValue('OtherRelatedInvoiceLines'),
}));
jest.mock('./ActionMenu', () => jest.fn().mockReturnValue('ActionMenu'));
jest.mock('./InvoiceLineInformation', () => jest.fn().mockReturnValue('InvoiceLineInformation'));
jest.mock('./ReceivingHistory', () => ({
  ReceivingHistory: jest.fn().mockReturnValue('ReceivingHistory'),
}));

const defaultProps = {
  invoiceLine,
  invoiceStatus: 'Approved',
  poLine: orderLine,
  currency: 'USD',
  closeInvoiceLine: jest.fn(),
  deleteInvoiceLine: jest.fn(),
  goToEditInvoiceLine: jest.fn(),
  tagsToggle: jest.fn(),
  vendorInvoiceNo: '1234',
  vendorCode: 'AMZN',
  otherRelatedInvoiceLines: [{ id: 'invoiceLineId' }],
};
const renderInvoiceLineDetails = (props = defaultProps) => render(
  <InvoiceLineDetails {...props} />,
  { wrapper: MemoryRouter },
);

describe('InvoiceLineDetails', () => {
  beforeEach(() => {
    useExchangeCalculation.mockReturnValue({
      exchangedAmount: 100,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correct subtitle for invoice line', () => {
    renderInvoiceLineDetails();
    const subTitle = `${defaultProps.vendorInvoiceNo} - ${defaultProps.vendorCode}`;

    expect(screen.getByText(subTitle)).toBeInTheDocument();
  });

  describe('Actions', () => {
    beforeEach(() => {
      ActionMenu.mockClear();
    });

    describe('Edit', () => {
      it('should call goToEditInvoiceLine when edit action is pressed in ActionMenu', () => {
        const goToEditInvoiceLine = jest.fn();

        renderInvoiceLineDetails({
          ...defaultProps,
          goToEditInvoiceLine,
          poLine: undefined,
          invoiceLine: undefined,
        });

        act(() => {
          ActionMenu.mock.calls[0][0].onEdit();
        });

        expect(goToEditInvoiceLine).toHaveBeenCalled();
      });
    });

    describe('Delete', () => {
      it('should open confirmation modal when Delete invoice line action is called', () => {
        renderInvoiceLineDetails();

        act(() => {
          ActionMenu.mock.calls[0][0].onDelete();
        });

        expect(screen.getByText('ui-invoice.invoiceLine.delete.message')).toBeDefined();
      });

      it('should call deleteInvoiceLine when delete action is confirmed', async () => {
        const deleteInvoiceLine = jest.fn();

        renderInvoiceLineDetails({
          ...defaultProps,
          deleteInvoiceLine,
        });

        act(() => {
          ActionMenu.mock.calls[0][0].onDelete();
        });
        await user.click(screen.getByText('ui-invoice.invoiceLine.delete.confirmLabel'));

        expect(deleteInvoiceLine).toHaveBeenCalled();
      });
    });
  });

  describe('Shortcuts', () => {
    beforeEach(() => {
      HasCommand.mockClear();
      expandAllSections.mockClear();
      collapseAllSections.mockClear();
    });

    it('should call expandAllSections when expandAllSections shortcut is called', async () => {
      renderInvoiceLineDetails();

      act(() => {
        HasCommand.mock.calls[0][0].commands.find(c => c.name === 'expandAllSections').handler();
      });

      expect(expandAllSections).toHaveBeenCalled();
    });

    it('should call collapseAllSections when collapseAllSections shortcut is called', () => {
      renderInvoiceLineDetails();

      act(() => {
        HasCommand.mock.calls[0][0].commands.find(c => c.name === 'collapseAllSections').handler();
      });

      expect(collapseAllSections).toHaveBeenCalled();
    });

    it('should call goToEditInvoiceLine when edit shortcut is called', () => {
      const goToEditInvoiceLine = jest.fn();

      renderInvoiceLineDetails({ ...defaultProps, goToEditInvoiceLine });
      HasCommand.mock.calls[0][0].commands.find(c => c.name === 'edit').handler();

      expect(goToEditInvoiceLine).toHaveBeenCalled();
    });
  });
});
