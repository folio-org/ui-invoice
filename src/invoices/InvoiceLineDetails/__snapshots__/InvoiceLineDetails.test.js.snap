// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceLineDetails should render correct structure for invoice line 1`] = `
<DocumentFragment>
  <div>
    <section
      class="pane focusIndicator"
      id="pane-invoiceLineDetails"
    >
      <div
        aria-labelledby="paneHeaderpane-invoiceLineDetails-pane-title"
        class="paneHeader"
        data-test-pane-header="true"
        id="paneHeaderpane-invoiceLineDetails"
        tabindex="-1"
      >
        <div
          class="paneHeaderButtonsArea first"
        >
          <div
            class="paneMenu"
          >
            <button
              aria-label="stripes-components.closeItem"
              class="iconButton medium paneHeaderCloseIcon PaneHeaderIconButton paneHeaderIconButton"
              data-test-pane-header-dismiss-button="true"
              icon="times"
              type="button"
            >
              <span
                class="iconButtonInner mediumOffset PaneHeaderIconButton__inner"
                data-test-pane-header-dismiss-button="true"
              >
                <span>
                  Icon
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="paneHeaderCenter"
        >
          <div
            class="paneHeaderCenterInner"
            id="paneHeaderpane-invoiceLineDetails-pane-title"
          >
            <h2
              class="paneTitle"
              data-test-pane-header-title="true"
            >
              <span
                class="paneTitleLabel"
              >
                ui-invoice.invoiceLine.paneTitle.view
              </span>
            </h2>
          </div>
        </div>
        <div
          class="paneHeaderButtonsArea last"
        >
          <div
            class="paneMenu"
          >
            <div
              class="dropdown"
              data-pane-header-actions-dropdown="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="button primary marginBottom0 actionMenuToggle"
                data-test-pane-header-actions-button="true"
                type="button"
              >
                <span
                  class="inner"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
              <div
                style="display: none;"
              >
                <div
                  class="DropdownMenu"
                  style="left: 0px; display: block;"
                >
                  ActionMenu
                </div>
              </div>
            </div>
            <div
              class="paneMenu"
            >
              <button
                aria-label="stripes-acq-components.showTags"
                class="iconButton hasBadge medium"
                data-test-invoice-line-tags-action="true"
                icon="tag"
                id="clickable-show-tags"
                title="stripes-acq-components.showTags"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset"
                  data-test-invoice-line-tags-action="true"
                  title="stripes-acq-components.showTags"
                >
                  <span>
                    Icon
                  </span>
                  <span
                    class="badge default medium"
                  >
                    <span
                      class="label"
                    >
                      0
                    </span>
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="paneContent hasPadding"
        id="pane-invoiceLineDetails-content"
        tabindex="-1"
      >
        <div
          class="row end-xs"
        >
          <div
            class="col-xs-12"
          >
            <button
              class="button link bottomMargin0"
              data-tast-expand-button="true"
              type="button"
            >
              <span
                class="inner"
              >
                <strong>
                  stripes-components.collapseAll
                </strong>
              </span>
            </button>
          </div>
        </div>
        <div
          aria-multiselectable="true"
        >
          <section
            class="accordion hasSeparator"
            data-test-accordion-section="true"
            id="invoiceLineInformation"
          >
            <div
              class="headerWrapper"
              tabindex="-1"
            >
              <div
                class="header default"
              >
                <h3
                  class="headline size-medium margin-none font-weight-bold block"
                  data-test-headline="true"
                >
                  <button
                    aria-controls="accordion1"
                    aria-expanded="true"
                    class="defaultCollapseButton"
                    id="accordion-toggle-button-invoiceLineInformation"
                    type="button"
                  >
                    <span
                      class="headerInner"
                    >
                      <span
                        class="defaultHeaderIcon"
                      >
                        <span>
                          Icon
                        </span>
                      </span>
                      <div
                        class="labelArea"
                      >
                        ui-invoice.invoiceLineInformation
                      </div>
                    </span>
                  </button>
                </h3>
              </div>
            </div>
            <div
              class="content-wrap expanded"
              style="z-index: 1;"
            >
              <div
                aria-labelledby="accordion-toggle-button-invoiceLineInformation"
                class="content-region expanded"
                data-test-accordion-wrapper="true"
                id="accordion1"
                role="region"
              >
                InvoiceLineInformation
              </div>
            </div>
          </section>
          <section
            class="accordion hasSeparator"
            data-test-accordion-section="true"
            id="invoiceLineFundDistribution"
          >
            <div
              class="headerWrapper"
              tabindex="-1"
            >
              <div
                class="header default"
              >
                <h3
                  class="headline size-medium margin-none font-weight-bold block"
                  data-test-headline="true"
                >
                  <button
                    aria-controls="accordion2"
                    aria-expanded="true"
                    class="defaultCollapseButton"
                    id="accordion-toggle-button-invoiceLineFundDistribution"
                    type="button"
                  >
                    <span
                      class="headerInner"
                    >
                      <span
                        class="defaultHeaderIcon"
                      >
                        <span>
                          Icon
                        </span>
                      </span>
                      <div
                        class="labelArea"
                      >
                        ui-invoice.fundDistribution
                      </div>
                    </span>
                  </button>
                </h3>
              </div>
            </div>
            <div
              class="content-wrap expanded"
              style="z-index: 1;"
            >
              <div
                aria-labelledby="accordion-toggle-button-invoiceLineFundDistribution"
                class="content-region expanded"
                data-test-accordion-wrapper="true"
                id="accordion2"
                role="region"
              >
                FundDistributionView
              </div>
            </div>
          </section>
          <section
            class="accordion hasSeparator"
            data-test-accordion-section="true"
            id="invoiceLineAdjustments"
          >
            <div
              class="headerWrapper"
              tabindex="-1"
            >
              <div
                class="header default"
              >
                <h3
                  class="headline size-medium margin-none font-weight-bold block"
                  data-test-headline="true"
                >
                  <button
                    aria-controls="accordion3"
                    aria-expanded="true"
                    class="defaultCollapseButton"
                    id="accordion-toggle-button-invoiceLineAdjustments"
                    type="button"
                  >
                    <span
                      class="headerInner"
                    >
                      <span
                        class="defaultHeaderIcon"
                      >
                        <span>
                          Icon
                        </span>
                      </span>
                      <div
                        class="labelArea"
                      >
                        ui-invoice.adjustments
                      </div>
                    </span>
                  </button>
                </h3>
              </div>
            </div>
            <div
              class="content-wrap expanded"
              style="z-index: 1;"
            >
              <div
                aria-labelledby="accordion-toggle-button-invoiceLineAdjustments"
                class="content-region expanded"
                data-test-accordion-wrapper="true"
                id="accordion3"
                role="region"
              >
                AdjustmentsDetails
              </div>
            </div>
          </section>
          OtherRelatedInvoiceLinesReceivingHistory
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
